### PR TITLE
[IMP] l10n_cl: Electronic Invoicing eCommerce and POS

### PR DESCRIPTION
**Pull request by masi-odoo**
Adding the new functionality for V17 which let's the end client to decide the electronic document to be generated for their eCommerce purchase and POS.

eCommerce Photo folder: https://drive.google.com/drive/folders/17DNHC6o3A7Fzab61nrnlFqB78j7X-TLd?usp=drive_link
POS Photo folder: https://drive.google.com/drive/folders/17zrIsuhZ2IAWX6X30IJ1HdFF0531gB3_?usp=sharing

### CODE REVIEW

**Review by samueljlieber — CHANGES_REQUESTED** [review]
Hi @masi-odoo, thanks for creating this PR! I have some corrections and suggestions for improvement for the sections you added. 

Regarding the images, it appears the folder you linked contains images for a different PR. I think this PR can suffice without screenshots. However, please link the correct images if you would like them to be included 🙂

I will push up a second commit with my changes so you can review.

**Comment by samueljlieber** [comment]
Implemented my technical changes in https://github.com/odoo/documentation/commit/29219b32f12293af00571e6086de3245d4caddd7 and added images.

File: content/applications/finance/fiscal_localizations/chile.rst [hunk 1]
```diff
@@ -947,6 +949,213 @@ valid and a new section needed for customs.
...
+  purchase.
+
+Once all of the configurations are made for the Chilean :ref:`electronic invoice
+<chile/electronic-invoice>` flow, the following configurations are required for the eCommerce flow
+to be integrated.
+
+To configure your website to generate electronic documents during the sale process, go to
+:menuselection:`Website --> Configuration --> Settings --> Invoicing` and activate the
+:guilabel:`Automatic Invoice` feature. Activating this feature allows electronic documents to be
+automatically generated when the online payment is confirmed.
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/chile.rst:976]:
```suggestion
automatically generated when an online payment is confirmed.
```

File: content/applications/finance/fiscal_localizations/chile.rst [hunk 2]
```diff
@@ -947,6 +949,213 @@ valid and a new section needed for customs.
...
+
+.. image:: chile/website-configurations-ecommerce-chile.png
+   :align: center
+   :alt: Invoice Policy and Automatic Invoice configurations.
+
+Since an online payment needs to be confirmed for the *automatic invoice* feature to generate the
+document, a payment provider should be configured for the related website.
+
+.. note::
+   Review the :doc:`../payment_providers` documentation to know which are supported in Odoo and how
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/chile.rst:986]:
```suggestion
   Review the :doc:`../payment_providers` documentation for information on which payment providers are supported in Odoo, and how
```

File: content/applications/finance/fiscal_localizations/chile.rst [hunk 3]
```diff
@@ -947,6 +949,213 @@ valid and a new section needed for customs.
...
+   to configure them.
+
+It is also recommended to configure your products so they are able to be invoiced at the time of an
+online payment confirmation. To do so, go to :menuselection:`Website --> eCommerce --> Products` and
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/chile.rst:990]:
```suggestion
It is also recommended to configure your products so they are able to be invoiced when an online payment is confirmed. To do so, go to :menuselection:`Website --> eCommerce --> Products` and
```

File: content/applications/finance/fiscal_localizations/chile.rst [hunk 4]
```diff
@@ -947,6 +949,213 @@ valid and a new section needed for customs.
...
+:guilabel:`Ordered quantities`.
+
+.. image:: chile/ordered-quantities-product.png
+   :align: center
+   :alt: Invoice Policy configuration in Products.
+
+Invoicing flows
+~~~~~~~~~~~~~~~
+
+Clients from Chile will be able to select if they need an invoice or a ballot for their purchase in
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/chile.rst:1001]:
```suggestion
Clients from Chile will be able to select if they need an invoice or a ballot for their purchase with
```

File: content/applications/finance/fiscal_localizations/chile.rst [hunk 5]
```diff
@@ -947,6 +949,213 @@ valid and a new section needed for customs.
...
+an extra step added in the checkout process.
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/chile.rst:1002]:
```suggestion
an extra step added during the checkout process.
```

File: content/applications/finance/fiscal_localizations/chile.rst [hunk 6]
```diff
@@ -947,6 +949,213 @@ valid and a new section needed for customs.
...
+
+.. image:: chile/select-edi-docs-ecommerce.png
+   :align: center
+   :alt: Option for EDI Documents for clients.
+
+If the client selects the :guilabel:`Electronic Invoice` option, fiscal fields will be required to
+be filled; such as the :guilabel:`Activity Description`, the :guilabel:`Identification Number` and
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/chile.rst:1009]:
```suggestion
be filled out; such as the :guilabel:`Activity Description`, the :guilabel:`Identification Number` and
```

File: content/applications/finance/fiscal_localizations/chile.rst [hunk 7]
```diff
@@ -947,6 +949,213 @@ valid and a new section needed for customs.
...
+their :guilabel:`DTE Email`.
+
+.. image:: chile/fiscal-fields-invoice-ecommerce.png
+   :align: center
+   :alt: Fiscal fields required for an Invoice to be requested.
+
+If the client selects the :guilabel:`Electronic Receipts` option, they will be directed to the next
+step and the electronic document will be generated for the *Consumidor Final Anónimo* contact.
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/chile.rst:1017]:
```suggestion
step, and the electronic document will be generated for the *Consumidor Final Anónimo* contact.
```

File: content/applications/finance/fiscal_localizations/chile.rst [hunk 8]
```diff
@@ -947,6 +949,213 @@ valid and a new section needed for customs.
...
+
+.. note::
+   If a purchase through eCommerce requires an export, the customer will need to contact your
+   company to generate an electronic export invoice (*document type 110*), which can be done from
+   the *Accounting* app.
+
+Point of Sale electronic invoicing
+----------------------------------
+
+To install the :guilabel:`Chilean Module for Point of Sale` go to :menuselection:`Apps` and search
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/chile.rst:1030]:
```suggestion
To install the :guilabel:`Chilean Module for Point of Sale` go to the :menuselection:`Apps` application on the main Odoo dashboard, and search
```

File: content/applications/finance/fiscal_localizations/chile.rst [hunk 9]
```diff
@@ -947,6 +949,213 @@ valid and a new section needed for customs.
...
+
+.. image:: chile/pos-edi-module-chile.png
+   :align: center
+   :alt: l10n_cl POS EDI module.
+
+This module enables the following features and configurations to:
+
+- Generate electronic documents from the *Point of Sale* application.
+- Support for required fiscal fields for contacts created in the *Point of Sale* application.
+- Effectively let the final client decide the type of electronic document to be generated for their
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/chile.rst:1041]:
```suggestion
- Effectively lets the final client decide the type of electronic document to be generated for their
```

File: content/applications/finance/fiscal_localizations/chile.rst [hunk 10]
```diff
@@ -947,6 +949,213 @@ valid and a new section needed for customs.
...
+  purchase.
+- Print QR or 5 digit codes in tickets to access to electronic invoices.
+
+To configure contacts with the required fiscal information, review the :ref:`partner information
+<chile/partner-information>` section, or directly modify a contact from :menuselection:`Point of
+Sale --> Session --> Customers --> Details` to edit:
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/chile.rst:1047]:
```suggestion
<chile/partner-information>` section, or directly modify a contact. Navigate to :menuselection:`Point of
Sale --> Session --> Customers --> Details`, and edit any of the following fields:
```

File: content/applications/finance/fiscal_localizations/chile.rst [hunk 11]
```diff
@@ -947,6 +949,213 @@ valid and a new section needed for customs.
...
+- :guilabel:`Type Giro`
+- :guilabel:`DTE Email`
+- :guilabel:`RUT`
+
+.. image:: chile/fiscal-required-pos-session.png
+   :align: center
+   :alt: Contact with fiscal information created from POS.
+
+To configure the products, navigate to :menuselection:`Point of Sale --> Products --> Products` and
+select a product record. On the :guilabel:`Sales` tab in the product form, it is necessary to mark
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/chile.rst:1062]:
```suggestion
select a product record. In the :guilabel:`Sales` tab of the product form, it is necessary to mark
```

File: content/applications/finance/fiscal_localizations/chile.rst [hunk 12]
```diff
@@ -947,6 +949,213 @@ valid and a new section needed for customs.
...
+.. note::
+   If the client requests a credit note due to a return of their purchase, the credit note should be
+   made using the *Accounting* app. See the :doc:`credit notes and refunds
+   <../accounting/customer_invoices/credit_notes>` documentation for detailed instructions.
+
+Electronic receipts: specific customer
+**************************************
+
+When making a purchase as a specific user that does not request an electronic invoice, Odoo
+automatically selects the contact for the order as the :guilabel:`Consumidor Final Anónimo`, and
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/chile.rst:1107]:
```suggestion
When a specific user making a purchase does not request an electronic invoice, Odoo
automatically selects the contact for the order as the :guilabel:`Consumidor Final Anónimo`, and
```

File: content/applications/finance/fiscal_localizations/chile.rst [hunk 13]
```diff
@@ -947,6 +949,213 @@ valid and a new section needed for customs.
...
+
+For electronic receipts (not generated for the *Consumidor Final Anónimo*) and electronic invoices,
+it is possible to manage the process to return products sold in a :abbr:`POS (Point of Sale)` order
+by selecting the :guilabel:`Refund` button.
+
+.. image:: chile/refund-order.png
+   :align: center
+   :alt: Refund option in the POS application.
+
+Orders can be searched by status of the order or by contact and be selected for the refund to be
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/chile.rst:1145]:
```suggestion
Orders can be searched by the order status or contact, and be selected for the refund to be
```

**Review by larm-odoo — COMMENTED** [review]
Hi @samueljlieber - I finished my review. I noticed I could only check on the green text/new text but could not make any edits or suggestions to the text above or below it, so I couldn't check that part. If you need me to, just let me know (and please show me how I can do that ). Otherwise- here you go!

**Comment by samueljlieber** [comment]
Thank you for your helpful review @larm-odoo! I implemented your suggestions in https://github.com/odoo/documentation/commit/64fb31d3e5be81c0a8d5a5db2977e55ec4dc6ae2.

File: content/applications/finance/fiscal_localizations/chile.rst [hunk 14]
```diff
@@ -947,6 +949,214 @@ valid and a new section needed for customs.
...
+Clients from Chile will be able to select if they need an invoice or a ballot for their purchase
+with an extra step added during the checkout process.
+
+.. image:: chile/select-edi-docs-ecommerce.png
+   :align: center
+   :alt: Option for EDI Documents for clients.
+
+If the client selects the :guilabel:`Electronic Invoice` option, fiscal fields will be required to
+be filled out; such as the :guilabel:`Activity Description`, the :guilabel:`Identification Number`
+and their :guilabel:`DTE Email`.
```
Comment (jero-odoo) [inline content/applications/finance/fiscal_localizations/chile.rst:1010]:
```suggestion
be filled out, including the :guilabel:`Activity Description`, the :guilabel:`Identification Number`
and their :guilabel:`DTE Email`.
```

File: content/applications/finance/fiscal_localizations/chile.rst [hunk 15]
```diff
@@ -947,6 +949,214 @@ valid and a new section needed for customs.
...
+
+.. image:: chile/fiscal-fields-invoice-ecommerce.png
+   :align: center
+   :alt: Fiscal fields required for an Invoice to be requested.
+
+If the client selects the :guilabel:`Electronic Receipts` option, they will be directed to the next
+step, and the electronic document will be generated for the *Consumidor Final Anónimo* contact.
+
+Clients from other countries (not Chile), will have their electronic receipts automatically
```
Comment (jero-odoo) [inline content/applications/finance/fiscal_localizations/chile.rst:1019]:
```suggestion
Clients from countries other than Chile will have their electronic receipts automatically
```

File: content/applications/finance/fiscal_localizations/chile.rst [hunk 16]
```diff
@@ -947,6 +949,214 @@ valid and a new section needed for customs.
...
+`l10n_cl_edi_pos` and click the :guilabel:`Activate` button.
+
+.. image:: chile/pos-edi-module-chile.png
+   :align: center
+   :alt: l10n_cl POS EDI module.
+
+This module enables the following features and configurations to:
+
+- Generate electronic documents from the *Point of Sale* application.
+- Support for required fiscal fields for contacts created in the *Point of Sale* application.
```
Comment (jero-odoo) [inline content/applications/finance/fiscal_localizations/chile.rst:1041]:
```suggestion
- Support the required fiscal fields for contacts created in the *Point of Sale* application.
```

**Review by jero-odoo — APPROVED** [review]
hey @samueljlieber great job! Only a few suggestions but otherwise everything looks good!

**Review by samueljlieber — APPROVED** [review]
Thank you @jero-odoo, I implemented your suggestions in https://github.com/odoo/documentation/commit/6302aa53d503de7daec0b14947d9e5397714081c!

I am approving this PR now that it has two peer reviews, and a technical review by me.

@masi-odoo can you please [review the built documentation](https://runbot137.odoo.com/runbot/static/build/54947148-17-0/logs/build/html/applications/finance/fiscal_localizations/chile.html) one last time? Thank you!

**Review by samueljlieber — APPROVED** [review]
This PR is good to go @StraubCreative 👍

**Review by toaa-odoo — APPROVED** [review]
I changed a few words and cleaned up the screenshots (we don't do red arrows and squares). Sorry for the delay :)

**Review by StraubCreative — APPROVED** [review]
Seems good. Thanks everyone 👍 

@robodoo r+

**Pull request closed without merging**